src/main/java/com/staff_management_system/com/database_connection_package/HourlyEmployeeDataAccess.java

```diff
@@ -0,0 +1,266 @@
+
+    /**
+     * @param c establish the database connection
+     */
+
+    public void initialize(Connection c) {
+        try {
+            aConnection = c;
+            aStatement = aConnection.createStatement();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * close the database connection
+     */
+    public void terminate() {
+        try {
+            aStatement.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    /**
+     * retrieve the database connection
+     *
+     * @param id the id for employee
+     * @return anHourlyEmployee
+     * @throws NotFoundException, InvalidNameException, InvalidIdException, InvalidPasswordException
+     */
+    public HourlyEmployee retrieve(int id) throws NotFoundException, InvalidNameException, InvalidIdException, InvalidPasswordException {
+        anHourlyEmployee = null;
+        try {
+            aStatement = aConnection.createStatement();
+            // execute the SQL query statement
+            ResultSet rs = aStatement.executeQuery("SELECT * FROM tblhourlyemployee WHERE id=" + id);
+
+            // next method sets cursor & returns true if there is data
+
+            if (rs.next()) { // extract the data
+                anHourlyEmployee.setId(rs.getInt("id"));
+                anHourlyEmployee.setPassword(rs.getString("password"));
+                anHourlyEmployee.setFirstName(rs.getString("firstname"));
+                anHourlyEmployee.setLastName(rs.getString("lastname"));
+                anHourlyEmployee.setRole(rs.getString("role"));
+                anHourlyEmployee.setStartDate(rs.getDate("startdate"));
+                anHourlyEmployee.setSin(rs.getString("sin"));
+                anHourlyEmployee.setPhoneNumber(rs.getLong("phonenumber"));
+                anHourlyEmployee.setEmailAddress(rs.getString("emailaddress"));
+                anHourlyEmployee.setAddress(rs.getString("address"));
+                anHourlyEmployee.setHourly(rs.getFloat("hourly"));
+                anHourlyEmployee.setEnabled(rs.getBoolean("enabled"));
+                anHourlyEmployee.setDepartment(rs.getString("department"));
+
+
+                try {
+                    anHourlyEmployee = new HourlyEmployee(id, password, firstName, lastName, role, startDate, sin, phoneNumber,
+                            emailAddress, address, hourly, enabled, department);
+                } catch (InvalidEmployeeDataException e) {
+                    System.out.println("Attempt to retrieve an HourlyEmployee that does exist (Id: " + id + ")\n" + "HourlyEmployee record with id " + id + "retrieved from the database");
+                }
+
+            } else // nothing was retrieved
+            {
+                throw (new NotFoundException("Attempt to retrieve an HourlyEmployee that does not exist (Id: " + id + ") \n" + "HourlyEmployee with if of " + id + " not found in the database."));
```

Comment:
you do not need () around new NotFoundException line

Reply:
Will fix it.